plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -30,11 +31,12 @@
 from UM.i18n import i18nCatalog
 catalog = i18nCatalog("cura")
 
-class CuraEngineBackend(Backend):
+class CuraEngineBackend(QObject, Backend):
     ##  Starts the back-end plug-in.
     #
     #   This registers all the signal listeners and prepares for communication
     #   with the back-end in general.
+    #   CuraEngineBackend is exposed to qml as well.
     def __init__(self):
```

Comment:
The QObject constructor is `QObject(QObject* parent = nullptr)`. This means that if you do not pass a parent instance, it will be unparented and not automatically destroyed when its parent gets destroyed. Since we are dealing with Python here, it is not strictly necessary to give a QObject a parent.

However, if there is a chance that a QObject can survive past the lifetime of the Python wrapper, it *should* have a parent as that prevents Python/PyQt from destroying the Qt/C++ side of the object when the Python side is destroyed. This is quite relevant for things that are created from QML, less so for things created from Python. In general, however, it is best to be safe and implement __init__ of a QObject subclass as `__init__(self, parent = None)`.